src/ifcopenshell-python/ifcopenshell/util/unit.py

```diff
@@ -624,3 +622,83 @@ def format_length(
     elif unit_system == "metric":
         rounded_val = round(value / precision) * precision
         return f"{rounded_val:.{decimal_places}f}"
+
+
+def get_base_type_name(
+        content_type: ifcopenshell.ifcopenshell_wrapper.named_type | ifcopenshell.ifcopenshell_wrapper.type_declaration) -> ifcopenshell.ifcopenshell_wrapper.type_declaration | None:
+    cur_decl = content_type
+    while hasattr(cur_decl, "declared_type") is True:
+        cur_decl = cur_decl.declared_type()
+        if hasattr(cur_decl, "name") is False:
+            continue
+        if cur_decl.name() == "IfcLengthMeasure":
+            return cur_decl
+
+    if isinstance(cur_decl, ifcopenshell.ifcopenshell_wrapper.aggregation_type):
+        res = cur_decl.type_of_element()
+        cur_decl = res.declared_type()
+        if hasattr(cur_decl, "name") and cur_decl.name() == "IfcLengthMeasure":
+            return cur_decl
+        while hasattr(cur_decl, "declared_type") is True:
+            cur_decl = cur_decl.declared_type()
+            if hasattr(cur_decl, "name") is False:
+                continue
+            if cur_decl.name() == "IfcLengthMeasure":
+                return cur_decl
+
+    return None
+
+
+def convert_file_units(ifc_file: ifcopenshell.file, target_units: str) -> ifcopenshell.file:
```

Comment:
This probably applies to other APIs as well, but should we copy over the header values?

Reply:
Ah, that never occured to me. Do we have any code examples for copying the header values somewhere else?

Reply:
In https://github.com/IfcOpenShell/IfcOpenShell/pull/4125/commits/9335c8a0c8c9cbc2fc9056cec79dd0bdd10c7350 I changed the "copy" method to a more direct copy of the entire file instead of going element by element.

`file_patched = ifcopenshell.file.from_string(ifc_file.wrapped_data.to_string())`

Which I believe will include header information